tests/pass-dep/concurrency/apple-futex.rs

```diff
@@ -149,6 +149,64 @@ fn wait_wake() {
     t.join().unwrap();
 }
 
+fn wait_wake_multiple() {
```

Comment:
This looks like it could depend quite a bit on schedule non-determinism?

Might be good to make the entire test use `-Zmiri-preemption-rate=0` like we do for many concurrency tests; then `yield` can be used to explicitly control which thread gets to run when. That avoids having to sleep so much here as well.